sdcm/nemesis.py

```diff
@@ -1688,6 +1692,42 @@ def disrupt_nodetool_enospc(self, sleep_time=30, all_nodes=False):
                     finally:
                         clean_enospc_on_node(target_node=node, sleep_time=sleep_time)
 
+    def disrupt_end_of_quota_nemesis(self, sleep_time=30):
+        """
+        Nemesis flow
+        ---------------------
+        1. Enable quota on the node.
+        2. Check disk usage.
+        3. Define quota size same as disk usage + 1GB.
+        4. Approach end of quota in a loop with fallocate file that takes 90% of quota size each iteration.
+        5. Wait for end of quota message to appear in the log - "Disk quota exceeded" / Except I/O Error if happens.
+        6. Remove the sparse files.
+        7. Remove the quota by setting it to 0.
+        8. Restart scylla server.
+        9. Verify scylla is up.
+        """
+
+        node = self.target_node
+        if self._is_it_on_kubernetes():
+            raise UnsupportedNemesis("Skipping nemesis for kubernetes")
+
+        result = node.remoter.run('cat /proc/mounts')
+        if '/var/lib/scylla' not in result.stdout:
+            raise UnsupportedNemesis("Scylla doesn't use an individual storage, skip end of quota test")
+
+        enable_quota_on_node(node)
+        quota_enabled = verify_quota_is_enabled_on_node(node)
+        if not quota_enabled:
+            raise QuotaConfigurationFailure("Failed to configure quota on the cluster")
+
+        with ignore_disk_quota_exceeded_errors(node=node):
+            quota_size = configure_quota_on_node_for_scylla_user(node=node)
+            verify_quota_is_configured_for_scylla_user(node=node)
```

Comment:
if this fails, then quota on node will persist.
One proposal is to use try..catch or better to make `configure_quota_on_node_for_scylla_user` to be context manager which removes quota upon `__exit__`